Parser/asdl.py

```diff
@@ -64,34 +64,45 @@ def __init__(self, name, fields=None):
     def __repr__(self):
         return 'Constructor({0.name}, {0.fields})'.format(self)
 
+class Quantifier:
+    class _Optional: pass
+    class _Sequence: pass
+    OPTIONAL = _Optional()
+    SEQUENCE = _Sequence()
+
 class Field(AST):
-    def __init__(self, type, name=None, seq=False, opt=False):
+    def __init__(self, type, name=None, quantifiers=None):
```

Comment:
Is there a need to support more than two qualifiers? and/or shouldn't we check the consistency of those qualifiers as well?

Reply:
Currently there are no more than two quantifiers on any type, as per my change to Python.asdl.

I'm just implementing the EBNF grammar described at the top of the asdl.py file. And I don't think this files does any amount of extra validation so far anyway.